### PR TITLE
fix GEARPUMP-109: add wordcount Java/Scala and DSL/low level API examples

### PR DESCRIPTION
**Pull request by whjiang**

### CODE REVIEW

**Comment by manuzhang** [comment]
could you help to fix this ?

```
- _site/gearpump-internals.html
  *  linking to internal hash #what-happen-when-executor-crash that does not exist (line 252)
- _site/get-gearpump-distribution.html
  *  internally linking to /download.html, which does not exist (line 144)
htmlproof 2.6.1 | Error:  HTML-Proofer found 2 failures!
```

**Comment by manuzhang** [comment]
honestly, the title "write your 1st gearpump app" is not as straightforward as "examples". What do you think ?

File: docs/dev-write-1st-app.md [hunk 1]
```diff
@@ -14,23 +14,131 @@ Repository and library dependencies can be found at [Maven Setting](maven-settin
...
+Gearpump supports two level APIs:
 
-An application is a Directed Acyclic Graph (DAG) of processors. In the wordcount example, We will firstly define two processors `Split` and `Sum`, and then weave them together.
+1. Low level API, which is more similar to Akka programming, operating on each event. The API document can be found at [Low Level API Doc](http://www.gearpump.io/releases/latest/api/scala/index.html#io.gearpump.streaming.package).
+
+2. High level API (aka DSL), which is operating on streaming instead of individual event. The API document can be found at [DSL API Doc](http://www.gearpump.io/releases/latest/api/scala/index.html#io.gearpump.streaming.dsl.package).
+
+And both APIs have their Java version and Scala version.
+
+So, before you writing our first Gearpump application, you need to decide which API to use and which language to use.
```
Comment (manuzhang) [inline docs/dev-write-1st-app.md]: "you writing our first" => "you write your first"

File: docs/dev-write-1st-app.md [hunk 2]
```diff
@@ -14,23 +14,131 @@ Repository and library dependencies can be found at [Maven Setting](maven-settin
...
+
+### Define Processor(Task) class and Partitioner class
+
+An application is a Directed Acyclic Graph (DAG) of processors. In the wordcount example, We will firstly define two processors `Split` and `Sum`, and then weave them together.
+
+
 #### Split processor
 
-In the Split processor, we simply split a predefined text (the content is simplified for conciseness) and send out each split word to Sum.
+In the `Split` processor, we simply split a predefined text (the content is simplified for conciseness) and send out each split word to Sum.
```
Comment (manuzhang) [inline docs/dev-write-1st-app.md]: "Sum" => "`Sum`"

**Comment by whjiang** [comment]
Hi @manuzhang,

1. for the "/download.html" link issue, I have fixed in PR #2. 
2. for the rest, I will try to fix it.

File: docs/dev-write-1st-app.md [hunk 3]
```diff
@@ -55,13 +163,58 @@ object Split {
...
+
+</div>
+
+</div>
+
+Essentially, each processor consists of two descriptions:
+
+1. A `Task` to define the operation.
+
+2. A parallelism level to define the parallelism of this processor. I.e. this processor will have how many tasks running in parallel.
```
Comment (manuzhang) [inline docs/dev-write-1st-app.md]: => "A parallelism level to define the number of tasks of this processor to run in parallel."

**Comment by whjiang** [comment]
For the name "write your 1st Gearpump app", I prefer it over "example".

**Comment by whjiang** [comment]
I don't think the failure is related to this PR as my PR is only document related.

**Comment by manuzhang** [comment]
LGTM + 1

**Comment by manuzhang** [comment]
please rebase

**Comment by whjiang** [comment]
rebase done.

**Pull request merged**